src/raw/mod.rs

```diff
@@ -433,15 +420,29 @@ impl<T> RawTable<T> {
     /// Deallocates the table without dropping any entries.
     #[cfg_attr(feature = "inline-more", inline)]
     unsafe fn free_buckets(&mut self) {
-        let (layout, _) =
+        let (layout, ctrl_offset) =
             calculate_layout::<T>(self.buckets()).unwrap_or_else(|| hint::unreachable_unchecked());
-        dealloc(self.ctrl.as_ptr(), layout);
+        dealloc(self.ctrl.as_ptr().sub(ctrl_offset), layout);
+    }
+
+
+    /// Returns pointer to one past last element of data table.
+    #[cfg_attr(feature = "inline-more", inline)]
+    pub unsafe fn data_backwards(&self) -> NonNull<T> {
+        NonNull::new_unchecked(self.ctrl.as_ptr() as *mut T)
+    }
+
+    /// Returns pointer to start of data table.
+    #[cfg_attr(feature = "inline-more", inline)]
+    #[allow(dead_code)]
```

Comment:
Why allow dead code here? If it is only used with certain features then you should add the same `#[cfg(feature = "...")]` flags as where it is used.